### PR TITLE
Epic: Jovu Phase 2

### PR DESCRIPTION
**Pull request by yuval-hazaz**
Fixes: #8307

## PR Details

Adds the plugins API client on amplication-server to make the catalog available for Jovu

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-server/src/core/assistant/assistant.service.ts [hunk 1]
```diff
@@ -665,5 +674,58 @@ export class AssistantService {
...
+        (plugin) => omit(plugin, ["__typename"])
+      );
+      this.logger.debug(
+        `Chat: Plugins: ${JSON.stringify(plugins)}`,
+        null,
+        context
+      );
+      return plugins;
+    },
+    installPlugins: async (
```
Comment (overbit) [inline packages/amplication-server/src/core/assistant/assistant.service.ts:688]: we are missing the unit tests here. Try the edge case of `this.pluginCatalogService.getPluginWithLatestVersion` returning  `undefined`

File: packages/amplication-server/src/core/pluginCatalog/pluginCatalog.service.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      const plugin =
+        data.plugins && data.plugins.length ? data.plugins[0] : undefined;
+      if (plugin) {
+        const versions = plugin.versions.filter((version) => version.isLatest);
+        return {
+          ...plugin,
+          versions,
+        };
+      }
+      return undefined;
```
Comment (overbit) [inline packages/amplication-server/src/core/pluginCatalog/pluginCatalog.service.ts:71]: undefined shouldn't be valid as the return type is Promise<PluginCatalogItem>

File: packages/amplication-server/src/core/assistant/assistant.service.ts [hunk 3]
```diff
@@ -665,5 +674,58 @@ export class AssistantService {
             : "Entity",
       }));
     },
+    getPlugins: async (args: undefined, context: AssistantContext) => {
+      const plugins = (await this.pluginCatalogService.getPlugins()).map(
+        (plugin) => omit(plugin, ["__typename"])
```
Comment (overbit) [inline packages/amplication-server/src/core/assistant/assistant.service.ts:679]: keep the data manipulation close to the action. 
`PluginCatalogItem` doesn't contain `__typename` so this line should have not effect. A unit tests would have caught it.

File: packages/amplication-server/src/core/pluginCatalog/pluginCatalog.service.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
+import { Injectable } from "@nestjs/common";
+import { ApolloClient, InMemoryCache, gql } from "@apollo/client";
```
Comment (overbit) [inline packages/amplication-server/src/core/pluginCatalog/pluginCatalog.service.ts:2]: To avoid any React dependency, import the apollo client from the core instead.
```suggestion
import { ApolloClient, InMemoryCache, gql } from "@apollo/client/core";
```

File: packages/amplication-client/src/Assistant/AssistantMessage.tsx [hunk 5]
```diff
@@ -28,7 +30,14 @@ const AssistantMessage = ({ message, onOptionClick }: Props) => {
...
           <>
-            <Icon icon="ai" color={EnumTextColor.ThemeTurquoise} size="large" />
+            <img
+              src={jovu}
+              alt="jovu"
+              width={30}
+              height={30}
+              style={{ background: "white", borderRadius: "50%" }}
+            />
+            {/* <Icon icon="ai" color={EnumTextColor.ThemeTurquoise} size="large" /> */}
```
[the remark this answers is not in the document]
Comment (yuval-hazaz) [reply packages/amplication-client/src/Assistant/AssistantMessage.tsx:40]: we are still working on the logo, so we will get back to it soon anyway.

File: packages/amplication-server/src/core/pluginCatalog/pluginCatalog.service.ts (hunk 4 above)
Comment (yuval-hazaz) [reply packages/amplication-server/src/core/pluginCatalog/pluginCatalog.service.ts:2]: nice!

**Review by overbit — APPROVED** [review]
Approved but it's missing unit tests. All these new services are missing them.
